registry/custom_reg_fields.go

```diff
@@ -196,138 +190,48 @@ func checkUniqueOptions(options []registrationFieldOption) bool {
-		return
-	}
-	instUrl, err := url.Parse(instUrlStr)
-	if err != nil {
-		intError = errors.Wrap(err, "Bad server configuration. Registry.InstitutionsUrl is invalid")
-		extError = errors.New("Bad server configuration. Registry.InstitutionsUrl is invalid")
-		return
-	}
-	if !institutionsCache.Has(instUrl.String()) {
-		log.Info("Cache miss for institutions TTL cache. Will fetch from source.")
-		client := &http.Client{Transport: config.GetTransport()}
-		req, err := http.NewRequest(http.MethodGet, instUrl.String(), nil)
-		if err != nil {
-			intError = errors.Wrap(err, "Error making a request when fetching institution list")
-			extError = errors.New("Error when creating a request to fetch institution from remote url.")
-			return
-		}
-		res, err := client.Do(req)
-		if err != nil {
-			intError = errors.Wrap(err, "Error response when fetching institution list")
-			extError = errors.New("Error from response when fetching institution from remote url.")
-			return
-		}
-		if res.StatusCode != 200 {
-			intError = errors.New(fmt.Sprintf("Error response when fetching institution list with code %d", res.StatusCode))
-			extError = errors.New(fmt.Sprint("Error when fetching institution from remote url, remote server error with code: ", res.StatusCode))
-			return
-		}
-		resBody, err := io.ReadAll(res.Body)
-		if err != nil {
-			intError = errors.Wrap(err, "Error reading response body when fetching institution list")
-			extError = errors.New("Error read response when fetching institution from remote url.")
-			return
-		}
-		institutions := []Institution{}
-		if err = json.Unmarshal(resBody, &institutions); err != nil {
-			intError = errors.Wrap(err, "Error parsing response body when fetching institution list")
-			extError = errors.New("Error parsing response when fetching institution from remote url.")
-			return
-		}
-		institutionsCache.Set(instUrl.String(), institutions, ttlcache.DefaultTTL)
-		return institutions, nil, nil
-	} else {
-		institutions := institutionsCache.Get(instUrl.String())
-		// institutions == nil if key DNE or item has expired
-		if institutions == nil || institutions.Value() == nil {
-			intError = errors.New(fmt.Sprint("Fail to get institutions from internal TTL cache, key is nil or value is nil from key: ", instUrl))
-			extError = errors.New("Fail to get institutions from internal cache, key might be expired")
-			return
-		}
-		return institutions.Value(), nil, nil
-	}
-}
-
 // Initialize institutions list
 func InitInstConfig(ctx context.Context, egrp *errgroup.Group) error {
-	institutions := []Institution{}
+	institutions := []registrationFieldOption{}
 	if err := param.Registry_Institutions.Unmarshal(&institutions); err != nil {
 		log.Error("Fail to read Registry.Institutions. Make sure you had the correct format", err)
 		return errors.Wrap(err, "Fail to read Registry.Institutions. Make sure you had the correct format")
 	}
 
+	instRegIdx := -1
```

Comment:
Can you add a comment explaining how you use/modify this index? What happens in line 231 if `registrationFields` is empty and `instRegIdx` isn't updated from -1?